File: Nim/Kex.nim
from strutils import parseInt
import Kex_1sub
import os
import Player
import random
import times

var PLAYER_INITIAL_HP = 500
var PLAYER_DEFENSE = 30

type Kex* = ref object of RootObj

method execute*(this: Kex) =
    echo "冒険が今始まる！"
    var player = Player(hp: PLAYER_INITIAL_HP, defense: PLAYER_DEFENSE)
    var continue_game = 1
    var kills = 0
    randomize(getTime().toWinTime)

    while continue_game != 0:
        echo "現HP：", player.hp
        stdout.write "奥に進みますか？（１：奥に進む　０．帰る）＞"
        stdout.flushFile
        continue_game = parseInt(readLine(stdin))
        if continue_game != 0:
            var golem_lv = rand(4)
            attack_golem_ex(golem_lv, player)
            kills += 1
            if player.hp == 0:
                break
    
    echo "リ〇ミト！"
    echo "戦闘回数：", kills, "　残りHP：", player.hp

File: Dart/lib/Kex.dart
import 'dart:io';
import 'dart:math';
import 'package:Dart/Kex_1sub.dart';
import 'package:Dart/Player.dart';

class Kex {
  void Execute() {
    Random rand = Random();
    print('冒険が今始まる！');
    Player player = Player();
    int continueGame = 1;
    int kills = 0;
    while (continueGame != 0) {
      print('現HP：${player.Hp}');
      stdout.write('奥に進みますか？（１：奥に進む　０．帰る）＞');
      continueGame = int.parse(stdin.readLineSync());
      if (continueGame == 1) {
        int golemLv = rand.nextInt(5);
        player = AttackGolemEx(golemLv, player);
        kills++;
        if (player.Hp == 0) {
          break;
        }
      }
    }
    print('リ〇ミト！\n');
    print('戦闘回数：$kills　残りHP：${player.Hp}');
  }
}